### PR TITLE
change: move ./home/static to ./static

### PR DESCRIPTION
**Pull request by francisli**

### CODE REVIEW

**Comment by francisli** [comment]
@kevr I think you forgot to update links that reference the old paths...!

Screen Shot 2022-09-02 at 10 46 36 AM

I'll see if I can quickly patch this...

**Comment by francisli** [comment]
@kevr I can't figure out why the debug server in development can't serve the static files anymore...?

**Comment by kevr** [comment]
> @kevr I can't figure out why the debug server in development can't serve the static files anymore...?

The server has a bit of an unorthodox setup for static files.

Whenever I normally setup a fresh django project, there's a public folder that's used for "static files." However, the linkage to these is provided through django dynamically. The issue here is that it seems that this sort of django dynamic handling of the static files is completely missing, where a static directory was pretty much hardcoded to serve out the assets (which is why it was working previously).

On my local setup, I've gottan this patch to load up the staticfiles, but the only thing it really does is relocate them -- the problem still exists.

Perhaps, instead of moving this around, I should just take a look at reorganizing how statics are handled at all.

Check out the following doc: https://learndjango.com/tutorials/django-static-files (collectstatic section)

**Comment by kevr** [comment]
Ahh, I see. So, this project is using the `whitenoise` package which handles production staticfiles. So, it's a bit different than default django that I'm used to. I'll have to look into this more deeply.

**Comment by francisli** [comment]
I didn't generate the original project, but I didn't think that it had done anything custom to the setup- I think by default it puts static files in a per-project location, and it's a customization to move it to a common shared directory...?

Whitenoise is just for production serving, since we're on Heroku and I didn't want to deal with creating a custom buildpack setup with nginx, etc... it's also one of the recommended libraries in that link you shared...

**Comment by kevr** [comment]
Okay. Now I've got whitenoise working. However, we've got a bit of a detail here that we need to work out:

Using `whitenoise` to manage staticfiles means that, in both development and production, we must use `python manage.py collectstatic` to populate the correct staticfiles directory. The `whitenoise` app is missing from INSTALLED_APPS in our `settings.py`, which is responsible for part of the bootstrapping responsible for whitenoise' handling of static files. The storage was also not set.

Basically, our staticfiles handling right now is kind of a not completely defined whitenoise implementation, which breaks part of it (and i'm surprised it ever served a static file at all with the previous config, tbh).

We have two routes, as far as I can tell:
- We can choose `whitenoise` and fix up the doc/settings.py to work with it in all cases (devel, prod)
- We can split off between django's default staticfiles handling during development and using whitenoise in production

**Comment by francisli** [comment]
I think the intent was to use the Django servestatic in debug/development and whitenoise in production...

**Comment by kevr** [comment]
> I think the intent was to use the Django servestatic in debug/development and whitenoise in production...

Sounds good to me. Let me fix up settings.py for that, I'll open up a new PR regarding it today.

**Comment by francisli** [comment]
> > I think the intent was to use the Django servestatic in debug/development and whitenoise in production...
> 
> Sounds good to me. Let me fix up settings.py for that, I'll open up a new PR regarding it today.

Awesome... Heroku automatically runs collectstatic for us during deployment for production...

**Comment by kevr** [comment]
Alright, instead of rolling a fresh new PR, I've just added another commit to this PR which does the job of fixing django staticfiles through its `settings.py`.

So, come to find out, django core staticfiles middleware/views depend on the `settings.DEBUG` variable. If DEBUG is true, it will dynamically generate static responses based on the source directories. If DEBUG is false (in production), it expects that staticfiles are collected via `python manage.py collectstatic` (as is already done on heroku), and it then uses `whitenoise` to serve out those collected staticfiles.

**Comment by kevr** [comment]
This should be good to go now.
